modules/ce_google_docs/src/Controller/GoogleDocsController.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+namespace Drupal\ce_google_docs\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Drupal\node\Entity\Node;
+use GuzzleHttp\Client;
+
+/**
+ * Class GoogleDocsController.
+ */
+class GoogleDocsController extends ControllerBase {
+
+  /**
+   * To update a node.
+   *
+   * @param Symfony\Component\HttpFoundation\Request $request
+   *   HttpRequest with various parameters of a a node.
+   *
+   * @return Symfony\Component\HttpFoundation\JsonResponse
+   *   JsonResponse with Node's details.
+   */
+  public function update(Request $request) {
+    $config = \Drupal::config('ce_google_docs.settings');
+    if ($request->request->get('apiKey') != $config->get('api_key')) {
+      return new JsonResponse(['error' => 'Invalid API Key.']);
+    }
+
+    $body = $request->request->get('body');
+    if ($request->request->get('img_store') == 'drupal') {
+      preg_match_all('/< *img[^>]*src *= *["\']?([^"\']*)/i', $body, $result);
+      $result = $result[1];
+      foreach ($result as $src) {
+        $client = new Client();
+        $response = $client->request('GET', $src);
+        $name = str_replace(" ", "_", substr($response->getHeader('Content-Disposition')[0], 17, -1));
```

Comment:
I don't think its better to save the file with its name, there can be a possibility of having more images with the same name and this will overwrite the other images

Reply:
At line 39 `$replace = FILE_EXISTS_RENAME` will rename the file in case of another file with same name

Reply:
that's what we won't want, why to replace the image of some other post

Reply:
@anantjain6 Can you reply to @ankitjain28may question?

Reply:
I agree that image file with the same name should not get overwritten
That why at line 39  I have already added `$replace = FILE_EXISTS_RENAME`
If it found any file with repetitive name, let's say `name.png` then it will save it as `name_0.png`, next time `name_1.png` and so on ...

Reply:
@anantjain6 please check my pm on Slack.

Reply:
@anantjain6 There should be some unique name for every image you upload else it will keep uploading unnecessary images on the local storage. Every time I edit and publish, it will upload the new image for the editing. and if we are not storing the revision, images should be deleted.